unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor_v2.cs

```diff
@@ -0,0 +1,1613 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace DCL.Skybox
+{
+    public enum SkyboxEditorToolsParts
+    {
+        Timeline_Tags,
+        BG_Layer,
+        Ambient_Layer,
+        Avatar_Layer,
+        Fog_Layer,
+        Directional_Light_Layer,
+        Base_Skybox
+    }
+
+    public class RightPanelPins
+    {
+        public string name;
+        public SkyboxEditorToolsParts part;
+        public int baseSkyboxSelectedIndex;
+        public bool pinned;
+        public Vector2 scroll;
+    }
+
+    public class SkyboxEditorWindow_v2 : EditorWindow
+    {
+        private EditorToolMeasurements toolSize;
+
+        private List<SkyboxConfiguration> configurations;
+        private List<string> configurationNames;
+        private int selectedConfigurationIndex;
+        private int newConfigIndex;
+        private SkyboxConfiguration selectedConfiguration;
+
+        public bool isPaused;
+        public float timeOfTheDay;
+        public float lifecycleDuration = 1;
+
+        private Material selectedMat;
+        private Vector2 leftPanelScrollPos;
+        private Vector2 rightPanelScrollPos;
+        private Light directionalLight;
+        private bool creatingNewConfig;
+        private string newConfigName;
+        private bool overridingController;
+
+        private MaterialReferenceContainer.Mat_Layer matLayer = null;
+
+        private GUIStyle foldoutStyle;
+        private GUIStyle renderingMarkerStyle;
+        private GUIStyle configurationStyle;
+        private GUIStyle percentagePartStyle;
+        private GUIStyle transitioningBoxStyle;
+
+        private List<string> renderingOrderList;
+
+        private float leftPanelWidth;
+        private List<RightPanelPins> rightPanelPins = new List<RightPanelPins>() { new RightPanelPins { part = SkyboxEditorToolsParts.BG_Layer, name = "Background Layer" } };
+
+        [MenuItem("Window/Skybox Editor")]
+        static void Init()
+        {
+            SkyboxEditorWindow_v2 window = (SkyboxEditorWindow_v2)EditorWindow.GetWindow(typeof(SkyboxEditorWindow_v2));
+            window.minSize = new Vector2(500, 500);
+            window.Initialize();
+            window.Show();
+            window.InitializeWindow();
+        }
+
+        void Initialize() { toolSize = AssetDatabase.LoadAssetAtPath<EditorToolMeasurements>("Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/EditorToolSize.asset"); }
+
+        public void InitializeWindow() { EnsureDependencies(); }
+
+        private void Update()
+        {
+            if (selectedConfiguration == null)
+            {
+                return;
+            }
+
+            if (isPaused)
+            {
+                return;
+            }
```

Comment:
These conditions could be in 1 conditional

Reply:
Done